ComponentKit/Core/CKComponentController.h

```diff
@@ -79,4 +79,14 @@
  */
 - (BOOL)canPerformAction:(SEL)action withSender:(id)sender;
 
+
+/**
+ Initializes a controller with the first generation of component. You should not directly initialize a controller,
+ they are initialized for you by the infrastructure.
+ */
+- (instancetype)initWithComponent:(CKComponent *)component NS_DESIGNATED_INITIALIZER;
```

Comment:
Although I believe this is cleaner, I want to note for readers that **this is a breaking change with the public API**, which will force people updating ComponentKit to have to fix up their initializers in the subclasses.

The other alternative I considered is pushing onto a thread-local stack and keeping the component as-is to maintain backwards compatibility, but thought this was clearer. Open to other ideas here.